paddle/fluid/framework/new_executor/interpretercore.cc

```diff
@@ -224,7 +232,7 @@ paddle::framework::FetchList InterpreterCore::Run(
 }
 
 paddle::framework::FetchList InterpreterCore::Run(
-    const std::vector<std::string>& feed_names) {
+    const std::vector<std::string>& feed_names, bool need_fetch) {
```

Comment:
Since there is no fetch var in the running Program of conditional_block_op, whether the need_fetch parameter may not be required?

Reply:
For `ConditionalBlockOP` , it should return values when it does not need. This will make the variable fetched  deconstructed unexpectedly. So we need this parameter to tell executor whether it should return values.

Reply:
Maybe we can decide by whether the block has the `fetch op`? @zhiqiu